### ISSUE TITLE
Adapting the objprop code for ytvis 2021

### ISSUE THREAD

**Issue by Allamrahul** [anirudh-chakravarthy/ObjProp#6]
I have been trying to adapt your code to ytvis 2021 dataset. We are using the pretrained pth provided in the repository to generate the json file.

I am using the following command to generate the json: 
python3 tools/test_video.py configs/masktrack_rcnn_r50_fpn_1x_youtubevos_objprop.py pth_f/epoch_12.pth --out Inf_op.pkl --eval segm --save_path 'vis/' --show

When we submit the generated file on the evaluation server, it gives out a score which is in the order of 10^-4. 

Since we are using ytvis 2021, I adapted the CLAS
[config.txt](https://github.com/anirudh-chakravarthy/ObjProp/files/8430150/config.txt)
SES variable in YTVOSDataset class to suit the output classes of ytvis2021. I am attaching the config and inference log for your reference. Please let me know where I might be going wrong. Is there any other variable I should change to account for the change in the dataset
[config.txt](https://github.com/anirudh-chakravarthy/ObjProp/files/8430153/config.txt)
[output.txt](https://github.com/anirudh-chakravarthy/ObjProp/files/8430197/output.txt)
?

**Comment by Allamrahul** [comment]
An update: I tried with ytvis 19 dataset as well. Even with that, we get a score around 0.002. Am I missing something?

**Comment by anirudh-chakravarthy** [comment]
I haven't personally worked with the YTVIS 2021 dataset. On The YTVIS'19 dataset, the command looks right to me. One caveat maybe is that the output file is not named in the right format. From the evaluation website: `The submission file should be named as "results.json", and compressed without any subfolder.` So you'll need to rename the output file generated to `results.json` and then compress it.

**Comment by anirudh-chakravarthy** [comment]
I'm curious, how do the visualizations look?

**Comment by Allamrahul** [comment]
I modified the name of results file to results.json. The following is the command I am using for train (4 v100) and eval:

Train: python3 tools/train.py configs/masktrack_rcnn_r50_fpn_1x_youtubevos_objprop.py --gpus 4

Test: python3 tools/test_video.py configs/masktrack_rcnn_r50_fpn_1x_youtubevos_objprop.py pth_f/epoch_12.pth --out results.pkl --eval segm --save_path 'vis/' --show

I did the following for ytvis 19:
1. I used the pretrained model(from your repo) and evaluated on the validation set to generate the results.json file. I got 0.0022532913 score on the ytvis 19 dataset
2. I trained the model from scratch for 12 epochs and evaluated on the validation set to generate the json file. This did not make much difference either. 

The following is the structure of our dataset:

-train
-valid
     - Annotations
         -.....
     - JPEGImages
         -.......
-vis 
    - train.json
    - valid.json

The visualizations are decent. I am lost as to what I am doing wrong. Could you please point us in the right direction?

**Comment by anirudh-chakravarthy** [comment]
Hi,
 
Were you able to resolve this? If not, I can have a look at replicating it when I get some time later this week!

**Comment by Allamrahul** [comment]
We observed a strange issue. The --show parameter during evaluation is
resulting in the bad score. Without it, we get around 0.35. with it, the
score is pretty bad. Not sure why this is happening.

**Comment by anirudh-chakravarthy** [comment]
I see. That's weird, and probably something carried forward from the original repo. Nonetheless, thanks for letting me know about that!
